From ab210d6646099bf6c5cff8509ee7489a3c206581 Mon Sep 17 00:00:00 2001
From: srihars1 <sriharsha.govinda.raj@intel.com>
Date: Tue, 3 Feb 2015 12:41:44 +0530
Subject: [PATCH] EM/FW: Fixed charger online status update issue.

Description: During charger fault condition,  charging framework
is not setting charger online to 0 if charger is disabled which
results unexpected behavior. This patch will fix this issue.

Change-Id: I060fd0f46fcf1f350b514db5080ce0832c8ee4f0
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-941
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Signed-off-by: srihars1 <sriharsha.govinda.raj@intel.com>
---
 drivers/power/power_supply_charger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 889fe69..6f63232 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -658,7 +658,7 @@ static void update_sysfs(struct power_supply *psy)
 			if (!IS_PRESENT(chrgr_lst[cnt]))
 				continue;
 
-			update_charger_online(psy);
+			update_charger_online(chrgr_lst[cnt]);
 		}
 		/* set battery status */
 		if (set_battery_status(psy, get_battery_status(psy)))
@@ -679,7 +679,7 @@ static void update_sysfs(struct power_supply *psy)
 					 * forcefully cache the battery
 					 * properties
 					 */
-					cache_cur_batt_prop_force(psy);
+					cache_cur_batt_prop_force(psb);
 		}
 	}
 }
-- 
1.9.1

